web/src/test/java/org/openmrs/web/controller/LoginControllerTest.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.controller;
+
+import org.junit.Test;
+import org.openmrs.web.WebConstants;
+import org.openmrs.web.test.BaseWebContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ *
+ */
+public class LoginControllerTest extends BaseWebContextSensitiveTest {
+	
+	@Autowired
+	private LoginController controller;
+	
+	@Test
+	public void shouldSetTheRedirectAttribute() {
+		String redirectUrl = "www.openmrs.org/_HASHTAG_";
+		
+		MockHttpServletRequest mockRequest = new MockHttpServletRequest();
+		mockRequest.setParameter("redirect_url", redirectUrl);
+		
```

Comment:
Is this test about setting the redirect attribute or replacing _HASHTAG_ with # ?